### ISSUE TITLE
How on earth do you actually use cuttlefish?

### ISSUE THREAD

**Issue by theelous3** [mlandauer/cuttlefish#358]
## How on earth do you actually use cuttlefish?

Without anything to indicate what to do off the bat, I cloned the repo, hoping for the best.
I saw instructions for docker compose, so I installed docker and docker compose.

Once I installed docker/compose I cd'd in to the dockerfiles folder and ran the commands listed.

It didn't seem to succeed, but it didn't seem to fail either. Got a big traceback about :

```
rake aborted!
ActiveRecord::RecordInvalid: Validation failed: From domain doesn't have a DNS record configured correctly for cuttlefish._domainkey.foo.com
[... BLAH BLAH BLAH ...]
/usr/local/bundle/bin/bundle:23:in `load'
/usr/local/bundle/bin/bundle:23:in `<main>'
Tasks: TOP => db:setup => db:seed
(See full trace by running task with --trace)
```

Whatever, I said. We'll see how she goes. Opening localhost to :3000 I was greeted with... a replica of the site hosted at `cuttlefish.io`. Not what I was expecting. I was expecting some kind of frontend admin interface, perhaps listing where I should be sending mail or whatever. The screenshots on the project site seem to indicate this exists.

I `telnet localhost 2525` and I'm informed that the STMP server is there and happy, at least.

I visit localhost:1080 and I'm happy to see MailCatcher is chilling there. Good good.

*But where is my interface?*

Looking around the repo and grepping git log, I see signup is impossible and was removed in 2013. No clear route in the debugger worked to move the process forward. Even on the actual hosted site, this is the case.

I gave up and removed it from my system. I'm pretty sure my story is typical of interactions with cuttlefish.

TL;DR, looks like https://github.com/postalhq/postal is a better bet if you're not literally the opensource foundation of australia. Given the lack of interest in making this project something people can easily use, I think it not too bad form to link elsewhere.

**Comment by mlandauer** [comment]
@theelous3 There's a "Sign In" button at the top right of the page. If you're just setting up a development version there's some seed data populated including a default admin. This is now documented in the README. If you're running this in production and/or with an empty database it will prompt you to create the first admin when you try to sign in.

The bug you came across during running the seeding of the database (https://github.com/mlandauer/cuttlefish/issues/340) has also been fixed.

Otherwise, the tone of your comments wasn't particularly helpful or kind. Just worth remembering there are people at the other end of what you write.

**Comment by theelous3** [comment]
Firstly, please don't take what I initially wrote as an attack on you, nor my follow-up here as some desire to flamewar.

> Otherwise, the tone of your comments wasn't particularly helpful or kind. Just worth remembering there are people at the other end of what you write.

My intent was to give you a step by step idea of what is likely a typical user story with your project, as I would be a fairly typical user. Generally technically competent, some understanding of email - but not enough to go the whole way alone. I believe I achieved that, it just wasn't a very happy story and my post reflected that. (That and I thought perhaps the project was dead and some dust needed a wee kicking.)

Cuttlefish seems to have been in that condition for (?) years whilst also giving the impression it was in a much more usable state, given the (very pretty) website and claims of 

> Cuttlefish is a lovely, **_easy to set up_** transactional email server

Granted now that (seed data has been added? or it was already there and undocumented but now it's documented?) it's probably in better shape.

Thanks for adding the additional docs and docker command, and for fixing the db issue. Purely out of a love of your little cuttlefish logo, I'm going to give it another go :) Thanks for your time.

**Comment by cameronbraid** [comment]
Thanks for the pointer to https://github.com/postalhq/postal   that works really well
